cmd/osbuild-jobsite-builder/main.go

```diff
@@ -206,6 +208,42 @@ func (builder *Builder) HandlePopulate(w http.ResponseWriter, r *http.Request) e
 	if r.Method != "POST" {
 		return fmt.Errorf("Builder.HandlePopulate: unexpected request method")
 	}
+	storePath := path.Join(argBuildPath, "store")
+	err := os.Mkdir(storePath, 0755)
+	if err != nil {
+		return fmt.Errorf("Builder.HandlePopulate: failed to make store directory")
+	}
+
+	tarReader := tar.NewReader(r.Body)
+	for header, err := tarReader.Next(); err != io.EOF; header, err = tarReader.Next() {
+		if err != nil {
+			return fmt.Errorf("Builder.HandlerPopulate: failed to unpack sources")
+		}
+		switch header.Typeflag {
+		case tar.TypeDir:
+			if err := os.Mkdir(path.Join(storePath, header.Name), 0755); err != nil {
+				return fmt.Errorf("Builder.HandlerPopulate: unable to make dir in sources")
+			}
+		case tar.TypeReg:
+			fmt.Println("Making file: %v", path.Join(storePath, header.Name))
+			err = os.MkdirAll(filepath.Dir(path.Join(storePath, header.Name)), 0755)
+			if err != nil {
+				return fmt.Errorf("Builder.HandlerPopulate: unable to make directory %v under sources %v", filepath.Dir(path.Join(storePath, header.Name)), err)
+			}
+
+			file, err := os.Create(path.Join(storePath, header.Name))
+			defer file.Close()
+			if err != nil {
+				return fmt.Errorf("Builder.HandlerPopulate: unable to open file %s in sources", path.Join(storePath, header.Name))
+			}
+			if _, err := io.Copy(file, tarReader); err != nil {
+				return fmt.Errorf("Builder.HandlerPopulate: unable to write file in sources")
+			}
+
+		default:
+			return fmt.Errorf("Builder.HandlerPopulate: unexpected tar header type")
```

Comment:
probably good to include here what the unexpected type is/was for easier debugging/logging